homeassistant/components/xiaomi_miio/humidifier.py

```diff
@@ -208,44 +197,49 @@ def __init__(self, name, device, entry, unique_id, coordinator):
             self._max_humidity = 80
             self._humidity_steps = 10
 
-        self._attributes.update({attribute: None for attribute in AVAILABLE_ATTRIBUTES})
+        self._state = self.coordinator.data.is_on
+        self._attributes.update(
+            {
+                key: self._extract_value_from_attribute(self.coordinator.data, value)
+                for key, value in AVAILABLE_ATTRIBUTES.items()
+            }
+        )
+        self._target_humidity = self._attributes[ATTR_TARGET_HUMIDITY]
+        self._mode = self._attributes[ATTR_MODE]
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
 
     @callback
     def _handle_coordinator_update(self):
         """Fetch state from the device."""
-        state = self.coordinator.data
-        if not state:
-            return
-        _LOGGER.debug("Got new state: %s", state)
-        self._available = True
-        self._state = state.is_on
+        self._state = self.coordinator.data.is_on
         self._attributes.update(
             {
-                key: self._extract_value_from_attribute(state, value)
+                key: self._extract_value_from_attribute(self.coordinator.data, value)
                 for key, value in AVAILABLE_ATTRIBUTES.items()
             }
         )
+        self._target_humidity = self._attributes[ATTR_TARGET_HUMIDITY]
+        self._mode = self._attributes[ATTR_MODE]
         self.async_write_ha_state()
 
     @property
     def mode(self):
         """Return the current mode."""
-        if self._state:
-            return AirhumidifierOperationMode(self._attributes[ATTR_MODE]).name
-        return None
+        return AirhumidifierOperationMode(self._mode).name
 
     @property
     def target_humidity(self):
         """Return the target humidity."""
-        if self._state:
-            return (
-                self._attributes[ATTR_TARGET_HUMIDITY]
-                if AirhumidifierOperationMode(self._attributes[ATTR_MODE])
-                == AirhumidifierOperationMode.Auto
-                or AirhumidifierOperationMode.Auto.name not in self.available_modes
-                else None
-            )
-        return None
+        return (
+            self._target_humidity
+            if self._mode == AirhumidifierOperationMode.Auto.name
+            or AirhumidifierOperationMode.Auto.name not in self.available_modes
+            else None
```

Comment:
If the ternary operator expression spans more than one line we prefer a regular if/else block.